ios/FluentUI/Button/FluentButtonStyle.swift

```diff
@@ -0,0 +1,98 @@
+//
+//  Copyright (c) Microsoft Corporation. All rights reserved.
+//  Licensed under the MIT License.
+//
+
+import SwiftUI
+import UIKit
+
+/// ButtonStyle which configures the Button View according to its state and design tokens.
+public struct FluentButtonStyle: SwiftUI.ButtonStyle {
+    public init(style: ButtonStyle, size: ButtonSizeCategory) {
+        self.style = style
+        self.size = size
+        self.tokenSet = ButtonTokenSet(style: { style }, size: { size })
+    }
+
+    public func makeBody(configuration: Configuration) -> some View {
+        tokenSet.update(fluentTheme)
+        let isPressed = configuration.isPressed
+        let isDisabled = !isEnabled
+        let isFocused = isFocused
+
+        let cornerRadius = tokenSet[.cornerRadius].float
+        let shadowToken = (isDisabled || isFocused || isPressed) ? ButtonToken.shadowPressed : ButtonToken.shadowRest
+        let shadowInfo = tokenSet[shadowToken].shadowInfo
+
+        let foregroundColor: Color
+        let borderColor: Color
+        let backgroundColor: Color
+        if isDisabled {
+            foregroundColor = Color(tokenSet[.foregroundDisabledColor].uiColor)
+            borderColor = Color(tokenSet[.borderDisabledColor].uiColor)
+            backgroundColor = Color(tokenSet[.backgroundDisabledColor].uiColor)
+        } else if isPressed || isFocused {
+            foregroundColor = Color(tokenSet[.foregroundPressedColor].uiColor)
+            borderColor = Color(tokenSet[.borderPressedColor].uiColor)
+            backgroundColor = Color(tokenSet[.backgroundPressedColor].uiColor)
+        } else {
+            foregroundColor = Color(tokenSet[.foregroundColor].uiColor)
+            borderColor = Color(tokenSet[.borderColor].uiColor)
+            backgroundColor = Color(tokenSet[.backgroundColor].uiColor)
+        }
+
+        @ViewBuilder var backgroundView: some View {
+            if isFloatingStyle {
+                backgroundColor.clipShape(Capsule())
+            } else {
+                backgroundColor.clipShape(RoundedRectangle(cornerRadius: cornerRadius))
+            }
+        }
+
+        @ViewBuilder var overlayView: some View {
+            if borderColor != .clear {
```

Comment:
In the case that `borderColor` is `clear`, does this return as `nil` or some sort of optional view?

Reply:
That's correct, it'll be `nil`. The check is not strictly necessary (we can always just draw a clear border color) but it seems like a neat little saving to have one less thing rendered in common cases.